logdna/logdna.py

```diff
@@ -39,48 +44,50 @@ def __init__(self, key, options={}):
         self.index_meta = options.get('index_meta', False)
         self.flush_limit = options.get('flush_limit', defaults['FLUSH_BYTE_LIMIT'])
         self.flush_interval = options.get('flush_interval', defaults['FLUSH_INTERVAL'])
+        self.retry_interval_secs = options.get('retry_interval_secs', defaults['RETRY_INTERVAL_SECS'])
         self.tags = options.get('tags', [])
+        self.buf_retention_limit = options.get('buf_retention_limit', defaults['BUF_RETENTION_LIMIT'])
+        self.exception_flag = False
 
         if isinstance(self.tags, str):
             self.tags = [tag.strip() for tag in self.tags.split(',')]
         elif not isinstance(self.tags, list):
             self.tags = []
-
-
         self.setLevel(logging.DEBUG)
         self.lock = threading.RLock()
 
-        self.session = requests.Session()
-        retry = Retry(connect=self.connection_retries, backoff_factor=self.retry_backoff_factor)
-        adapter = HTTPAdapter(max_retries=retry)
-        self.session.mount('https://', adapter)
-
     def buffer_log(self, message):
         if message and message['line']:
             if len(message['line']) > self.max_length:
                 message['line'] = message['line'][:self.max_length] + ' (cut off, too long...)'
                 if self.verbose in ['true', 'debug', 'd']:
-                    logger.debug('Line was longer than %s chars and was truncated.', str(self.max_length))
-
-        self.buf_byte_length += sys.getsizeof(message)
+                    self.internalLogger.debug('Line was longer than %s chars and was truncated.', self.max_length)
 
         # Attempt to acquire lock to write to buf, otherwise write to secondary as flush occurs
         if not self.lock.acquire(blocking=False):
             self.secondary.append(message)
         else:
-            self.buf.append(message)
+            if self.buf_byte_length < self.buf_retention_limit:
+                self.buf_byte_length += sys.getsizeof(message)
+                self.buf.append(message)
+            else:
+                self.internalLogger.debug('The buffer size exceeded the limit: %s', self.buf_retention_limit)
+
             self.lock.release()
-            if self.buf_byte_length >= self.flush_limit:
+
+            if self.buf_byte_length >= self.flush_limit and self.exception_flag == False:
```

Comment:
PEP8 discourages comparing bool against `True` or `False` (see https://realpython.com/python-pep8/#programming-recommendations). So this is better `if self.buf_byte_length >= self.flush_limit and not self.exception_flag:`